code/game/turfs/simulated/floor/lava.dm

```diff
@@ -142,3 +142,100 @@
 
 /turf/simulated/floor/plating/lava/smooth/airless
 	temperature = TCMB
+
+/turf/simulated/floor/plating/lava/smooth/lava_land_surface/plasma
+	name = "liquid plasma"
+	desc = "A flowing stream of chilled liquid plasma. You probably shouldn't get in."
+	icon = 'icons/turf/floors/liquidplasma.dmi'
+	icon_state = "liquidplasma-255"
+	base_icon_state = "liquidplasma"
+	baseturf = /turf/simulated/floor/plating/lava/smooth/lava_land_surface/plasma
+
+	light_range = 3
+	light_power = 0.75
+	light_color = LIGHT_COLOR_PINK
+
+/turf/simulated/floor/plating/lava/smooth/lava_land_surface/plasma/examine(mob/user)
+	. = ..()
+	. += "<span class='info'>Some <b>liquid plasma<b> could probably be scooped up with a <b>container</b>.</span>"
+
+/turf/simulated/floor/plating/lava/smooth/lava_land_surface/plasma/attackby(obj/item/I, mob/user, params)
+	if(!I.is_open_container())
+		return ..()
+	if(!I.reagents.add_reagent("plasma", 10))
+		to_chat(user, "<span class='warning'>[I] is full.</span>")
+		return
+	to_chat(user, "<span class='notice'>You scoop out some plasma from the [src] using [I].</span>")
+
+/turf/simulated/floor/plating/lava/smooth/lava_land_surface/plasma/burn_stuff(AM)
+	. = FALSE
+	if(find_safeties())
+		return FALSE
+
+	var/thing_to_check = src
+	if(AM)
+		thing_to_check = list(AM)
+	for(var/thing in thing_to_check)
+		if(isobj(thing))
+			var/obj/O = thing
+			if(!O.simulated)
+				continue
+			if((O.resistance_flags & (LAVA_PROOF|INDESTRUCTIBLE)) || O.throwing)
+				continue
+			. = 1
+			if((O.resistance_flags & ON_FIRE))
+				continue
+			if(!(O.resistance_flags & FLAMMABLE))
+				O.resistance_flags |= FLAMMABLE //Even fireproof things burn up in lava
+			if(O.resistance_flags & FIRE_PROOF)
+				O.resistance_flags &= ~FIRE_PROOF
+			if(O.armor.getRating(FIRE) > 50) //obj with 100% fire armor still get slowly burned away.
+				O.armor = O.armor.setRating(fire_value = 50)
+			O.fire_act(10000, 1000)
+
+		if(isliving(thing))
```

Comment:
```suggestion
		if(isliving(thing))
```
this last if in the for loop can be turned into a guard clause with a continue

Reply:
Done!